src/allmydata/node.py

```diff
@@ -192,6 +200,18 @@ def read_config(self):
             if os.path.exists(self.config_fname):
                 raise
 
+    def write_config_file(self, name, value, mode="w"):
+        """
+        writes the given 'value' into a file called 'name' in the config
+        directory
+        """
+        fn = os.path.join(self._basedir, name)
+        try:
+            fileutil.write(fn, value, mode)
+        except EnvironmentError as e:
+            log.msg("Unable to write config file '{}'".format(fn))
+            log.err(e)
```

Comment:
writes it ... or not, apparently?  With logged and squashed failure?  I see that this is just duplicating old behavior so most likely it is appropriate for this to remain as-is.... but I have no idea why this is desirable behavior.  In fact, I'm pretty sure it is undesirable since it turns a visible fault into a hidden fault and I happen to know that in at least one case where this API is used (`announcement-seqnum`), the fault will escalate to a failure with user-facing consequences.

Reply:
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/2905